packages/cryptoplease/lib/features/swap/src/swap/process_swap_screen.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import '../../../../di.dart';
+import '../../../../ui/transfer_status/transfer_error.dart';
+import '../../../../ui/transfer_status/transfer_progress.dart';
+import '../../../../ui/transfer_status/transfer_success.dart';
+import '../swap.dart';
+import '../swap_route.dart';
+import 'swap_bloc.dart';
+import 'swap_repository.dart';
+
+class ProcessSwapScreen extends StatefulWidget {
+  const ProcessSwapScreen({
+    Key? key,
+    required this.route,
+  }) : super(key: key);
+
+  final SwapRoute route;
+
+  @override
+  State<ProcessSwapScreen> createState() => _ProcessSwapScreenState();
+}
+
+class _ProcessSwapScreenState extends State<ProcessSwapScreen> {
+  late Stream<Swap?> _swap;
+
+  @override
+  void initState() {
+    super.initState();
+    _swap = const Stream.empty();
+    context.read<SwapBloc>().add(SwapEvent.create(widget.route));
+  }
+
+  @override
+  Widget build(BuildContext context) => BlocListener<SwapBloc, SwapState>(
+        listenWhen: (prev, curr) => prev.isEmpty && curr.isNotEmpty,
+        listener: (_, state) {
+          setState(() {
+            _swap = sl<SwapRepository>().watch(state.single);
```

Comment:
Why is it single? What if there are multiple items in the set?

Can you make it in the same way as it's done with e.g. `IncomingSplitKeyPaymentScreen`?